library/std/src/Std/DurrHoyerLibrary.qs

```diff
@@ -0,0 +1,222 @@
+            set result += (1 <<< i);
+        }
+    }
+    return result;
+}
+
+// Oracle that marks elements less than the threshold through Most Signficant Bit comparision
+operation OracleLessThan(threshold : Int, inputQubits : Qubit[], auxQubit : Qubit) : Unit is Adj + Ctl {
+    // Convert the threshold to binary and compare
+    let thresholdBits = ConvertToBinary(threshold, Length(inputQubits));
+    for i in 0..Length(thresholdBits) - 1 {
+        if (thresholdBits[i] == Zero) {
+            //  Most Signficant Bit comparision, if There is a zero when the bits are compared we have something less than
+            X(inputQubits[i]); // Flip qubits that should be zero in the threshold
+        }
+    }
+
+    // Controlled-Z gate to flip the phase of the state if the element is less than the threshold
+    Controlled Z(inputQubits, auxQubit);
+
+    // Undo the X operations to revert qubits
+    for i in 0..Length(thresholdBits) - 1 {
+        if (thresholdBits[i] == Zero) {
+            X(inputQubits[i]);
+        }
+    }
+}
+
+// Oracle that marks elements more than the threshold through Most Signficant Bit comparision
+operation OracleMoreThan(threshold : Int, inputQubits : Qubit[], auxQubit : Qubit) : Unit is Adj + Ctl {
+    // Convert the threshold to binary and compare
+    let thresholdBits = ConvertToBinary(threshold, Length(inputQubits));
+    for i in 0..Length(thresholdBits) - 1 {
+        if (thresholdBits[i] == One) {
+            //  Most Signficant Bit comparision, if tbere is a one when the bits are compared we have something more than
+            X(inputQubits[i]); // Flip qubits that should be zero in the threshold
+        }
+    }
+
+    // Controlled-Z gate to flip the phase of the state if the element is less than the threshold
+    Controlled Z(inputQubits, auxQubit);
+
+    // Undo the X operations to revert qubits
+    for i in 0..Length(thresholdBits) - 1 {
+        if (thresholdBits[i] == One) {
+            X(inputQubits[i]);
+        }
+    }
+}
+
+// Diffusion operator (Grover's diffusion)
+operation DiffusionOperator(qubits : Qubit[]) : Unit {
+    ApplyToEach(H, qubits);
+    ApplyToEach(X, qubits);
+    Controlled Z(qubits[0..Length(qubits) - 2], qubits[Length(qubits) - 1]);
+    ApplyToEach(X, qubits);
+    ApplyToEach(H, qubits);
+}
+
+// Grover iteration with the oracle and diffusion operator for min
+operation GroverIterationMin(threshold : Int, inputQubits : Qubit[], auxQubit : Qubit) : Unit {
+    OracleLessThan(threshold, inputQubits, auxQubit);
+    DiffusionOperator(inputQubits);
+}
+
+// Grover iteration with the oracle and diffusion operator for max
+operation GroverIterationMax(threshold : Int, inputQubits : Qubit[], auxQubit : Qubit) : Unit {
+    OracleMoreThan(threshold, inputQubits, auxQubit);
+    DiffusionOperator(inputQubits);
+}
+
+// Dürr-Høyer for finding min or max algorithm
+operation DurrHoyerAlgorithm(list : Int[], nQubits : Int, type : String) : Int {
```

Comment:
Docs please since this would be a public facing function. You can refer tob existing code on how to write doc comments so that they get featured on Q# website

Reply:
I had added comments as required by guide, it looks like it got lost as I was developing. Adding required summary sections now.